src/GeocodeSharp/Google/GeocodeClient.cs

```diff
@@ -28,7 +28,7 @@ public GeocodeClient()
         public GeocodeClient(string apiKey)
         {
             _apiKey = apiKey;
-            _baseUrl = string.Format("https://maps.googleapis.com/maps/api/geocode/json?key={0}", Uri.EscapeDataString(_apiKey));
+            _baseUrl = string.Format("https://maps.googleapis.com/maps/api/geocode/json?key={0}&", Uri.EscapeDataString(_apiKey));
```

Comment:
When an api key is provided it was giving an error, when I checked Fiddler
to see what was being sent, the api key parameter had no ampersand
splitting it from the address parameter and Google was spitting back an
error about an invalid API key due to that. When I added the ampersand on
the api ctor it fixed it in my use case. I guess it'll give you the same
results when you initialize the client by passing an api key into the ctor?

It might be worth adding a status message that Google provides too, to make
it easier to see what errors are, eliminating the need to use something
like fiddler to see the exact problem Google is comaing about. I can add
that tomorrow, but can you confirm the API key issue?

On Tuesday, 21 April 2015, Brandon <notifications@github.com> wrote:

> In src/GeocodeSharp/Google/GeocodeClient.cs
> <https://github.com/bcuff/GeocodeSharp/pull/5#discussion_r28692560>:
>
> > @@ -28,7 +28,7 @@ public GeocodeClient()
> >          public GeocodeClient(string apiKey)
> >          {
> >              _apiKey = apiKey;
> > -            _baseUrl = string.Format("https://maps.googleapis.com/maps/api/geocode/json?key={0}", Uri.EscapeDataString(_apiKey));
> > +            _baseUrl = string.Format("https://maps.googleapis.com/maps/api/geocode/json?key={0}&", Uri.EscapeDataString(_apiKey));
>
> Actually I merged this too quickly. This doesn't fix anything. Did you
> have a problem initially?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bcuff/GeocodeSharp/pull/5/files#r28692560>.
>


-- 

[image: Hero115px.png]

ALEX FRAME  |  SOFTWARE ENGINEER  | HERO

Travel Technologies Pty Ltd, 131-135 Bourke St, Woolloomooloo NSW 2011
Australia

+61 2 9188 2020 | www.hero.travel